packaging/darwin/amazon-cloudwatch-agent-ctl

```diff
@@ -192,7 +175,7 @@ cwa_config() {
      if [ "${config_location}" = "${ALL_CONFIG}" ]; then
           rm -rf "${JSON_DIR}"/*
      else
-          runDownloaderCommand="${CMDDIR}/config-downloader --output-dir ${JSON_DIR} --download-source ${config_location} --mode ${param_mode} --config ${COMMON_CONIG} --multi-config ${multi_config}"
+          runDownloaderCommand="${CMDDIR}/config-downloader --output-dir ${JSON_DIR} --download-source ${config_location} --mode ${mode} --config ${COMMON_CONIG} --multi-config ${multi_config}"
```

Comment:
I would prefer fail fast since it relates to control script if the mode is invalid and besides, I have not seen any recover script after error is found though. Another thing that bothers me besides your questions: if the mode is invalid or somehow the config does not work on that particular mode, can we set mode to `auto` automatically to get the right mode for agent?

Reply:
True. The log message are missing the `config.ModeAuto`. Will change the log messages.

Reply:
No I have not removed the validation in the ctl script. It has a duplicated validation script. The flow would be like this:
1. User inputs their values, if they are not input for flag --mode , the default would be `auto`. If they input besides `auto, ec2, onPremise `, it would [cause an error](https://github.com/aws/amazon-cloudwatch-agent/pull/417/files#). This is the first validation 
2. After the corresponding action such as `fetch-config`, it will go through the `cwa_config` function and mets a second validation, which is the similar to the first one. 
So I don't see any reason to duplicate the validation if it does not use else where besides the ctl.